Backport patches for rdma cm patches

Backport for ipv6_ib_mc_map() for kernel v2.6.24 and earlier.

Signed-of-by: David Wilder <dwilder@us.ibm.com>
---
diff -Naurp ofa_kernel.orig/drivers/infiniband/core/cma.c ofa_kernel/drivers/infiniband/core/cma.c
--- ofa_kernel.orig/drivers/infiniband/core/cma.c	2009-12-08 16:13:37.000000000 -0600
+++ ofa_kernel/drivers/infiniband/core/cma.c	2009-12-08 16:19:09.000000000 -0600
@@ -2774,7 +2774,7 @@ static void cma_set_mgid(struct rdma_id_
 		/* IPv6 address is an SA assigned MGID. */
 		memcpy(mgid, &sin6->sin6_addr, sizeof *mgid);
 	} else if ((addr->sa_family == AF_INET6)) {
-		ipv6_ib_mc_map(&sin6->sin6_addr, dev_addr->broadcast, mc_map);
+		ipv6_ib_mc_map(&sin6->sin6_addr, mc_map);
 		if (id_priv->id.ps == RDMA_PS_UDP)
 			mc_map[7] = 0x01;	/* Use RDMA CM signature */
 		*mgid = *(union ib_gid *) (mc_map + 4);


